docs/rfc/0004_component_groups_scale_view.md

```diff
@@ -0,0 +1,299 @@
+    groupReplicas: # Group and their desired replicas. Once a group is owned by some scale view, then another scale view
+    # would not be able to grab the lock again. It's achieved with the atomic update.
+    # If a group and its desired replicas is listed below, then any changes to the target group's replicas will be rejected
+    # unless it equals to the replicas here (hence locks that group). This should be guaranteed by the webhook. 
+    - name: 'enduring'
+      replicas: 1
+    - name: 'spot'
+      replicas: 3
+```
+
+Validating webhook:
+
+- If the webhook observes scaleViews field, it must check the replicas field of each target groups and reject any change
+  not matches.
+
+This is a compatible change and should not affect the RisingWave's workflow.
+
+## `RisingWaveComponentScaleView` Webhooks
+
+![RisingWaveComponentScaleView Webhooks](./images/scale_view_webhooks.png)
+
+## `RisingWaveComponentScaleView` Controller
+
+![RisingWaveComponentScaleView Controller](./images/scale_view_controller_workflow.png)
+
+## Use Cases
+
+Case 1. Multi-AZ frontend and balanced scale
+
+```yaml
+apiVersion: risingwave.singularity-data.com/v1alpha1
+kind: RisingWave
+metadata:
+  name: multi-az
+spec:
+  components:
+    frontend:
+      groups:
+      - name: az-1
+        replicas: 1
+      - name: az-2
+        replicas: 1
+  # ....
+---
+apiVersion: risingwave.risingwavelabs.com/v1alpha1
+kind: RisingWaveComponentScaleView
+metadata:
+  name: multi-az-frontend-sv
+spec:
+  targetRef:
+    name: multi-az
+    component: frontend
+  scalePolicy:
+  - group: az-1
+  - group: az-2
+```
+
+When scaling with the `multi-az-frontend-sv`, the replicas of group `az-1` and `az-2` will increase/decrease
+alternately.
+
+```shell
+# Results in 2 replicas in az-1 and 2 replicas in az-2
+kubectl scale --replicas=4 risingwavecomponentscaleview/multi-az-frontend-sv
+```
+
+Case 2. Compactor auto scales onto spot instances
+
+```yaml
+apiVersion: risingwave.risingwavelabs.com/v1alpha1
+kind: RisingWavePodTemplate
+metadata:
+  name: spot-toleration
+template:
+  metadata:
+    labels:
+      risingwave/spot: "1"
+  spec:
+    containers: [ ]
+    tolerations:
+    - key: spot
+      operator: Exists
+      effect: NoSchedule
+---
+apiVersion: risingwave.singularity-data.com/v1alpha1
+kind: RisingWave
+metadata:
+  name: spot-c
+spec:
+  components:
+    compactor:
+      groups:
+      - name: normal
+        replicas: 1
+      - name: spot
+        replicas: 0
+        podTemplate: spot-toleration
+  # ....
+---
+apiVersion: risingwave.risingwavelabs.com/v1alpha1
+kind: RisingWaveComponentScaleView
+metadata:
+  name: spot-c-compactor-sv
+spec:
+  targetRef:
+    name: spot-c
+    component: compactor
+  scalePolicy:
+  - group: normal
+    priority: 1
+    constraints:
+      max: 2
+  - group: spot
+```
+
+```shell
+# Results in 1 replicas in normal and 0 replicas in spot
+kubectl scale --replicas=1 risingwavecomponentscaleview/spot-c-compactor-sv
```

Comment:
As in the previous question, can we use `kubectl  scale` to scale a specific  group?

Reply:
Answered above.